crates/btc-relay/src/lib.rs

```diff
@@ -482,6 +482,21 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    // TODO: wrap sla update via staked-relayers pallet
+    fn _store_block_header_and_update_sla(
+        relayer: T::AccountId,
+        raw_block_header: RawBlockHeader,
+    ) -> DispatchResult {
+        if let Err(err) = Self::_store_block_header(relayer.clone(), raw_block_header) {
+            if err != DispatchError::from(Error::<T>::DuplicateBlock) {
+                return Err(err);
+            }
+        }
+
+        ext::sla::event_update_relayer_sla::<T>(&relayer, ext::sla::RelayerEvent::BlockSubmission)?;
```

Comment:
I don't think there is anything preventing relayers from reporting the last 100 blocks to get perfect SLA, is there? We should only allow it for the newest block. And we could consider giving a lower sla increase for those that report the duplicate, to give them a little bit of incentive to be fast